stackrox/stackrox
# ROX-22046: Paladin Cloud integration form

## Description

Paladin Cloud integration form allows to configure integration with Paladin Cloud instances. This is new type of `Cloud Source` integrations, which will be extended with `OCM` integration type next release.
Three fields are required:
1. Name for the integration
2. Paladin Cloud endpoint
3. Token to access Paladin Cloud

For now we do not implement `Test` functionality either in the backend or frontend.

## Checklist

If any of these don't apply, please comment below.

## Testing Performed

### Here I tell how I validated my change

1. CI TBD
2. Manual test:
* Connect local frontend to infra [instance](https://infra.rox.systems/cluster/id-02-07-paladin) with cloud sources enabled
* Check that create/update/delete/view functionality work as expected
* Check that `updateCredentials` functionality works properly: credentials are modified when checkbox is checked, otherwise credentials are not sent to the backend


<img width="1777" alt="Screenshot 2024-02-08 at 18 02 10" src="https://github.com/stackrox/stackrox/assets/78353299/7e921a53-8225-4268-ae56-bc7347063208">
<img width="1777" alt="Screenshot 2024-02-08 at 18 02 31" src="https://github.com/stackrox/stackrox/assets/78353299/2f5461e6-f446-40f7-85da-8b863bc5b298">
<img width="1785" alt="Screenshot 2024-02-08 at 18 03 37" src="https://github.com/stackrox/stackrox/assets/78353299/e2d396f5-4a55-425a-855a-f57cfbdcaf22">
<img width="1780" alt="Screenshot 2024-02-08 at 18 03 49" src="https://github.com/stackrox/stackrox/assets/78353299/52896cc2-d448-43ac-af48-bd45deb629a0">
